API/Controllers/TournamentsController.cs

```diff
@@ -15,17 +15,33 @@ namespace API.Controllers;
 public class TournamentsController(ITournamentsService tournamentsService) : Controller
 {
     /// <summary>
-    /// List all tournaments
+    /// Get all tournaments which fit an optional request query
     /// </summary>
+    /// <param name="requestQuery">The optional request query filter</param>
     /// <remarks>Will not include match data</remarks>
-    /// <response code="200">Returns all tournaments</response>
+    /// <response code="200">Returns all tournaments which fit the request query</response>
+    /// <response code="401">If a request is made to fetch all tournaments without
+    /// the system claim</response>
+    /// <returns>All tournaments if the <param name="requestQuery">requestQuery</param> is null,
+    /// else returns all tournaments which fit the query
     [HttpGet]
-    [Authorize(Roles = OtrClaims.System)]
```

Comment:
If the intention was to only auth on system clients then why did you remove the attribute

Reply:
So, the intention here is to have a public endpoint (this should even be `AllowAnonymous` now that I think about it) that returns paginated tournaments. However, at least in my head, in order to retain the "get all" functionality, I opted to have `requestQuery` be optional. If the `requestQuery` is null and the authorized user is a system user, then we return all tournament data at once.

I did this because there is a conflict on `/tournaments GET` - I am unsure what the best solution is for having both a "filtered" get and an actual "get me everything" endpoint (at least from a naming perspective).

Reply:
I just don't see why we need a "get it all at once" endpoint available at all, it sounds like a terrible idea. The amount of data you are loading into memory and then transferring over network in one transaction by doing that is absurd. The processor doesn't even interact with the API anymore (not to mention it doesnt pull this data to begin with) and that is the only use case I can think of. Pagination should be the only way we support getting batches of entities IMO

Reply:
`/GET` `/tournaments` should be a means by which you get a batch of information about tournaments themselves. If one wants to get the entire tournament's store of match data, they should take a result from `/GET` `/tournaments` and request it from `/GET` `/tournaments/{id}`

We **SHOULD**:
- Allow granular filtering (by name, ruleset, verification status, processing status, submitter)
- Include compact DTOs for the submitter and verifier

We **SHOULDNT**:
- Include deep navigational data (matches, games, scores, beatmaps, etc)

Reply:
Yes all of these things are what this change tries to accomplish. Should both the GET all and GET (with filter) live under the same endpoint (how it is now)? If not, what should the naming scheme be?

Reply:
> I just don't see why we need a "get it all at once" endpoint available at all, it sounds like a terrible idea. The amount of data you are loading into memory and then transferring over network in one transaction by doing that is absurd. The processor doesn't even interact with the API anymore (not to mention it doesnt pull this data to begin with) and that is the only use case I can think of. Pagination should be the only way we support getting batches of entities IMO

That isn't the behavior though. This filter is for pagination. Without the filter, it acts as a standard `ListAsync`. So, if the filter is present (which it must be for any non-system web request), it will also be paginated.

Reply:
And yes because the processor doesn't use the API, I see no use case for a system-level client permission throughout the entire application. So if we want to just completely nuke that capability altogether (as I see no use case for a `/tournaments GET` returning all tournaments), then we can. I only kept this functionality as we tend to have a `/entity GET` which returns everything. 

Bigger question: Why have `ListAsync()` style endpoints at all?

Reply:
> That isn't the behavior though. This filter is for pagination. Without the filter, it acts as a standard ListAsync. So, if the filter is present (which it must be for any non-system web request), it will also be paginated.

I see, I didn't really understand what you were going for but I get it. What I did for the matches endpoint was just enforce the binding of `page` and `limit` so that the endpoint is pagination-only.

> I see no use case for a system-level client permission throughout the entire application.

I agree, we can probably axe that at this point

> Bigger question: Why have ListAsync() style endpoints at all?

`ListAsync()` is still the preferred naming scheme for this style of endpoint (see [here](https://cloud.google.com/apis/design/standard_methods#list)). I just think we were at a bit of a disconnect as to what it actually meant in the first place :P

Reply:
I will enforce the bindings of Page / PageSize here and remove the system client check for returning all tournaments.